src/bundle/Controller/User/ProfileEditController.php

```diff
@@ -0,0 +1,140 @@
+use Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException;
+use Ibexa\Contracts\Core\Repository\LanguageService;
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\Repository\PermissionResolver;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\UserService;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Contracts\Core\Repository\Values\User\User;
+use Ibexa\Core\Base\Exceptions\UnauthorizedException as CoreUnauthorizedException;
+use Symfony\Component\Form\Form;
+use Symfony\Component\HttpFoundation\Request;
+
+final class ProfileEditController extends Controller
+{
+    private Repository $repository;
+
+    private UserService $userService;
+
+    private LocationService $locationService;
+
+    private UserProfileConfigurationInterface $configuration;
+
+    private PermissionResolver $permissionResolver;
+
+    private LanguageService $languageService;
+
+    private ActionDispatcherInterface $userActionDispatcher;
+
+    private GroupedContentFormFieldsProviderInterface $groupedContentFormFieldsProvider;
+
+    public function __construct(
+        Repository $repository,
+        UserService $userService,
+        LocationService $locationService,
+        UserProfileConfigurationInterface $configuration,
+        PermissionResolver $permissionResolver,
+        LanguageService $languageService,
+        ActionDispatcherInterface $userActionDispatcher,
+        GroupedContentFormFieldsProviderInterface $groupedContentFormFieldsProvider
+    ) {
+        $this->repository = $repository;
+        $this->userService = $userService;
+        $this->locationService = $locationService;
+        $this->configuration = $configuration;
+        $this->permissionResolver = $permissionResolver;
+        $this->languageService = $languageService;
+        $this->userActionDispatcher = $userActionDispatcher;
+        $this->groupedContentFormFieldsProvider = $groupedContentFormFieldsProvider;
+    }
+
+    /**
+     * @return \Ibexa\ContentForms\User\View\UserUpdateView|\Symfony\Component\HttpFoundation\Response
+     */
+    public function editAction(Request $request, ?string $languageCode)
+    {
+        $user = $this->userService->loadUser($this->permissionResolver->getCurrentUserReference()->getUserId());
+        if (!$this->isUserProfileAvailable($user)) {
+            throw $this->createNotFoundException();
+        }
+
+        if (!$this->permissionResolver->canUser('user', 'selfedit', $user)) {
+            throw new CoreUnauthorizedException('user', 'selfedit', ['userId' => $user->getUserId()]);
+        }
+
+        $languageCode ??= $user->contentInfo->mainLanguageCode;
+
+        $data = (new UserUpdateMapper())->mapToFormData($user, $user->getContentType(), [
+            'languageCode' => $languageCode,
+            'filter' => static fn (Field $field): bool => $field->fieldTypeIdentifier !== 'ezuser',
```

Comment:
know is not ideal but maybe we could at least use `\Ibexa\Core\Repository\UserService::USER_FIELD_TYPE_NAME`

Reply:
What about https://github.com/ibexa/core/pull/309 ?